### PR TITLE
No Warning appears when a Dapp sets a really high Fees for a tx, potentially loosing all user funds

### LINKED ISSUE #6361: [Bug] No Warning appears when a Dapp sets a really high Fees for a tx, potentially loosing all user funds
**Describe the bug**
When a Dapp triggers a transaction/contract interaction, it sets the proposed fees maxFeePerGas maxPriorityFeePerGas and gasLimit. MetaMask takes the values from the Dapp as they are and displayes Site suggested meaning that it's the Dapp who suggested the fees. Despite of that, if the Dapp sets a high value, user can loose all its money and there is no warning on MM side.

**Screenshots**

Screenshot from 2023-05-10 12-21-26

**To Reproduce**
_Steps to reproduce the behavior_

1. Clone test dapp https://github.com/MetaMask/test-dapp
2. Set some high fee values on lines 637 and 638
```
maxFeePerGas: '0x38D7EA4C68000',
maxPriorityFeePerGas: '0x38D7EA4C68000',
```
5. Start test dapp: npm start
6. Go to localhost:9011
7. Connect MM
8. Click Send EIP1559
9. See suggested fees and no warning

**Expected behavior**
Same approach as in Extension, see discussion here https://github.com/MetaMask/metamask-extension/issues/17313

> As a solution, whenever the fee is equal or higher than our market suggestion, we should use the same warning color for the aggressive gas pricing option 

`This gas fee is much higher than the other options available.`

**Smartphone (please complete the following information):**
 - Device: Pixel 6
 - OS: android
 - App Version: `v6.6.0`

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by segun**
**Description**

When a Dapp triggers a transaction/contract interaction, it sets the proposed fees maxFeePerGas maxPriorityFeePerGas and gasLimit. MetaMask takes the values from the Dapp as they are and displayes Site suggested meaning that it's the Dapp who suggested the fees. Despite of that, if the Dapp sets a high value, user can loose all its money and there is no warning on MM side.

**Screenshots/Recordings**

**Before**

**After**

Screenshot 2023-05-10 at 12 50 31

**Issue**

Fixes #6361 

**Checklist**

* [x] There is a related GitHub issue
* [x] Tests are included if applicable
* [x] Any added code is fully documented

### CODE REVIEW

File: app/components/UI/TransactionReview/TransactionReviewEIP1559/index.test.tsx [hunk 1]
```diff
@@ -21,15 +19,12 @@ const initialState = {
...
-      </Provider>,
-    );
-    expect(wrapper).toMatchSnapshot();
+  it('should match snapshot', async () => {
+    const container = renderWithProvider(<TransactionReviewEIP1559 />, {
+      state: initialState,
+    });
+    expect(container).toMatchSnapshot();
   });
 });
```
Comment (jpuri) [inline app/components/UI/TransactionReview/TransactionReviewEIP1559/index.test.tsx:30]: 👍

**Review by jpuri — DISMISSED** [review]

**Comment by jpuri** [comment]
PR looks good. @segun please add label `unit test coverage confirmed` for all mobile PRs that have unit test coverage.

**Review by jpuri — DISMISSED** [review]

**Review by blackdevelopa — DISMISSED** [review]

**Comment by seaona** [comment]
hi @segun I'm seeing the following error when I build this branch, could you take a look? :pray: 

image

**Comment by bschorchit** [comment]
This should be the popover content: https://github.com/MetaMask/metamask-extension/issues/17313#issuecomment-1406824617

**Comment by holantonela** [comment]
> I believe the warning tooltip should also be next to Max fee instead of next to the time estimation similar to how it's on extension
Agreed.

**Comment by seaona** [comment]
A couple of small comments @segun 

## Color of time message
I can see how selecting the aggressive option, displays the time `Likely in 15 seconds) in black instead of in green as it used to do.

image


**Expected**
image



## Tooltip position of time message
In the case of  Low priority, I think that the tooltip should be on the left (indicating high time) instead of on the right (indicating high fees).
I won't be picky about it just want to confirm it's acceptable by design / product @bschorchit so we can merge it
image

File: app/components/UI/TimeEstimateInfoModal/index.js [hunk 2]
```diff
@@ -15,6 +15,8 @@ const TimeEstimateInfoModal = ({ timeEstimateId, isVisible, onHideModal }) => (
         ? strings('times_eip1559.warning_low_title')
         : timeEstimateId === AppConstants.GAS_TIMES.UNKNOWN
         ? strings('times_eip1559.warning_unknown_title')
+        : timeEstimateId === AppConstants.GAS_TIMES.VERY_LIKELY
```
[the remark this answers is not in the document]
Comment (segun) [reply app/components/UI/TimeEstimateInfoModal/index.js:18]: @bschorchit @seaona

[the remark this answers is not in the document]
Comment (bschorchit) [reply app/components/UI/TimeEstimateInfoModal/index.js:18]: oh, I'll need someone walking me through the meaning of this change 😅

[the remark this answers is not in the document]
Comment (bschorchit) [reply app/components/UI/TimeEstimateInfoModal/index.js:18]: ooh, sounds a positive fix to me then :)

**Comment by seaona** [comment]
Changes look good @segun ! I can see the issue fixed! Great work :fire:

File: ios/Podfile.lock [hunk 3]
```diff
@@ -874,4 +874,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 74e503a275b4ed097cb01c171027882e9cdc699d
 
-COCOAPODS: 1.12.0
+COCOAPODS: 1.11.3
```
Comment (jpuri) [inline ios/Podfile.lock:877]: 🔪

**Review by jpuri — DISMISSED** [review]
LGTM, please undo changes in [ios/Podfile.lock](https://github.com/MetaMask/metamask-mobile/pull/6358/files#diff-b7d85c35701918f25c9b2c418249f9b66fd9b067676e295a9eff493b73ee1752) before merging.

File: app/components/UI/EditGasFee1559Update/index.tsx [hunk 4]
```diff
@@ -669,7 +669,10 @@ const EditGasFee1559Update = ({
               </Text>
               <View style={styles.labelTextContainer}>
                 <Text
-                  green={timeEstimateColor === 'green'}
+                  green={
+                    timeEstimateColor === 'green' ||
+                    timeEstimateId === 'very_likely'
```
Comment (blackdevelopa) [inline app/components/UI/EditGasFee1559Update/index.tsx:674]: Same here
```suggestion
                    timeEstimateId === AppConstants.GAS_TIMES.VERY_LIKELY
```

File: ios/Podfile.lock [hunk 5]
```diff
@@ -777,7 +777,7 @@ EXTERNAL SOURCES:
 
 SPEC CHECKSUMS:
   boost: a7c83b31436843459a1961bfd74b96033dc77234
-  Branch: 74cc856025984f691833c8fa332834ac38a0cf4e
+  Branch: 4ac024cb3c29b0ef628048694db3c4cfa679beb0
```
Comment (blackdevelopa) [inline ios/Podfile.lock:780]: 🪓

**Review by blackdevelopa — DISMISSED** [review]
LGTM. Just small changes

**Review by jpuri — APPROVED** [review]

**Review by blackdevelopa — APPROVED** [review]

**Pull request merged**